File: table-value_count.py
import pandas as pd
import numpy as np


# Membuat DataFrame contoh
df = pd.DataFrame({
    'Nama': ["John", "Doe", "Jane", "Smith", "John", np.nan, "Bari", "Jan"],
    'Usia': [30, 25, 35, 40, 30, 34, np.nan, 35]
})
print(df)
print('\n\n')

print('nama-nama')
print(df.index)
print(df.columns)
print(df.values)


# print('Data tambahan')
# tabel = df['Nama'].value_counts()
# print(tabel)

# hasil = df['Nama'].value_counts(normalize=True)
# print(hasil)

# print(tabel.index)
# print()
# print(tabel.values)
# print()
# print(hasil.index)
# print()
# print(hasil.values)
# print()
# print(hasil.values.sum().round())


# print('\n\n\n\n\n')

# print(df.value_counts())
# print(df[['Nama', 'Usia']].value_counts())


# tabel = df.value_counts(normalize=True)
# print(tabel)

# hasil = df[['Nama', 'Usia']].value_counts(normalize=True)
# print(hasil)



# print(tabel.index)
# print()
# print(tabel.values)
# print()
# print(hasil.index)
# print()
# print(hasil.values)
# print()
# print(hasil.values.sum().round())

File: table-value_count.r
# Membuat DataFrame contoh
df <- data.frame(
    Nama = c("John", "Doe", "Jane", "Smith", "John", NA, "Bari", "Jan"),
    Usia = c(30, 25, 35, 40, 30, 34, NA, 35)
)



tabel <- table(df$Nama)
print(tabel)

hasil <- prop.table(table(df$Nama))
print(hasil)

print(names(tabel))
print(as.vector(tabel))
print(names(hasil))
print(as.vector(hasil))

print(sum(as.vector(hasil)))
cat('\n\n\n\n\n\n\n')


tabel <- table(df)
print(tabel)

tabel <- table(df[c('Nama', 'Usia')])
print(tabel)

hasil <- prop.table(table(df))
print(hasil)



hasil <- prop.table(table(df[c('Nama', 'Usia')]))
print(hasil)


cat('\n\n')

print(dim(tabel))
print(dimnames(tabel))
print(as.vector(tabel))
print(dim(hasil))
print(dimnames(hasil))
print(as.vector(hasil))

print(sum(as.vector(hasil)))
